bee-ledger/src/workers/snapshot/download.rs

```diff
@@ -1,41 +1,200 @@
+        };
+
+        if let Some(delta_header) = self.delta_header.as_ref() {
+            if delta_header.network_id() != wanted_network_id {
+                warn!(
+                    "delta snapshot network ID does not match ({} != {}): {}",
+                    delta_header.network_id(),
+                    wanted_network_id,
+                    self.urls.delta()
+                );
+                return false;
+            };
+
+            if self.full_header.sep_index() > delta_header.sep_index() {
+                warn!(
+                    "full snapshot SEP index is bigger than delta snapshot SEP index ({} > {}): {}",
+                    self.full_header.sep_index(),
+                    delta_header.sep_index(),
+                    self.urls.full()
+                );
+                return false;
+            }
 
-pub(crate) async fn download_snapshot_file(
-    path: &Path,
-    download_urls: impl Iterator<Item = &str>,
+            if self.full_header.sep_index() != delta_header.ledger_index() {
+                warn!(
+                    "full snapshot SEP index does not match the delta snapshot ledger index ({} > {}): {}",
+                    self.full_header.sep_index(),
+                    delta_header.ledger_index(),
+                    self.urls.full()
+                );
+                return false;
+            }
+        }
+
+        true
+    }
+}
+
+async fn gather_source_information(
+    download_delta: bool,
+    urls: &'_ DownloadUrls,
+) -> Result<SourceInformation<'_>, Error> {
+    let full_header = download_snapshot_header(urls.full()).await?;
+    let delta_header = if download_delta {
+        Some(download_snapshot_header(urls.delta()).await?)
+    } else {
+        None
+    };
+
+    Ok(SourceInformation {
+        urls,
+        full_header,
+        delta_header,
+    })
+}
+
+pub(crate) async fn download_latest_snapshot_files(
+    wanted_network_id: u64,
+    full_snapshot_path: &Path,
+    delta_snapshot_path: Option<&Path>,
+    download_urls: &[DownloadUrls],
 ) -> Result<(), Error> {
+    let download_delta = delta_snapshot_path.is_some();
+
+    let all_sources = join_all(
+        download_urls
+            .iter()
+            .map(|source| gather_source_information(download_delta, source)),
+    )
+    .await;
+
+    let mut available_sources = all_sources
+        .into_iter()
+        .flatten()
+        .filter(|source| source.is_consistent(wanted_network_id))
+        .collect::<Vec<SourceInformation>>();
+
+    // Sort all available sources so that the freshest is at the end.
+    available_sources.sort_by(|a, b| {
+        let cmp_full_index = a.full_header.ledger_index().cmp(&b.full_header.ledger_index());
+        let cmp_delta_index = a
+            .delta_header
+            .as_ref()
+            .map(|x| x.ledger_index())
+            .cmp(&b.delta_header.as_ref().map(|x| x.ledger_index()));
+        cmp_full_index.then(cmp_delta_index)
+    });
```

Comment:
@thibault-martinez Does the sorting make sense? In Hornet, I think they only sort on the `ledger_index` of the full snapshot.

Reply:
Let's do the same as hornet